src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -936,10 +960,50 @@ public void setHealth(double health) {
         }
     }
 
+    @Override
+    public void damageBlock(double amount, Block block, DamageCause cause) {
+        if (!canTakeDamage() || !canTakeDamage(cause)) {
+            return;
+        } else {
+            noDamageTicks = maximumNoDamageTicks;
+        }
+
+        // armor damage protection
+        // formula source: http://minecraft.gamepedia.com/Armor#Damage_Protection
+        double defensePoints = getAttributeManager().getPropertyValue(Key.KEY_ARMOR);
+        double toughness = getAttributeManager().getPropertyValue(Key.KEY_ARMOR_TOUGHNESS);
+        amount = amount * (1 - Math.min(20.0,
+                Math.max(defensePoints / 5.0,
+                        defensePoints - amount / (2.0 + toughness / 4.0))) / 25);
+
+        // fire event
+        EntityDamageByBlockEvent event = EventFactory.getInstance().onEntityDamage(
```

Comment:
It's really non-ideal that we have to do this. Could we extract some of the damage code into two parts, so that both the entity and block pathway can call it appropriately?